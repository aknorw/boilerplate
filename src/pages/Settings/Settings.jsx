import React from 'react'
import PropTypes from 'prop-types'
import { Helmet } from 'react-helmet'
import { connect } from 'react-redux'

import { setFoo } from 'services/basic/actions'

// This container does not follow the rule about DOM markup
// This is intentional as this is only temporary

import Button from 'components/Button'

export const Settings = ({ handleButtonClick }) => (
  <div>
    <Helmet title="Settings" />
    <h1>
      Settings
    </h1>
    <Button onClick={handleButtonClick}>
      Change foo from Settings
    </Button>
  </div>
)

Settings.propTypes = {
  handleButtonClick: PropTypes.func.isRequired,
}

const mapDispatchToProps = dispatch => ({
  handleButtonClick: () => dispatch(setFoo()),
})

export default connect(null, mapDispatchToProps)(Settings)
