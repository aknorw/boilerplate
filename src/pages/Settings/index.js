import { createAsyncComponent } from 'utils'

export default createAsyncComponent({
  loader: () => import('./Settings' /* webpackChunkName: 'settings' */),
})
