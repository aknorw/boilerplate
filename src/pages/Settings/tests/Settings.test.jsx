import React from 'react'
import { shallow } from 'enzyme'

import { Settings } from '../Settings'

const mockFn = jest.fn()
console.error = jest.fn()

describe('<Settings />', () => {
  it('should be defined', () => {
    expect(Settings).toBeDefined()
  })
  it('should log 1 error when no props are passed', () => {
    shallow(<Settings />)
    expect(console.error).toHaveBeenCalledTimes(1)
  })
  it('should render correctly', () => {
    const tree = shallow(<Settings handleButtonClick={mockFn} />)
    expect(tree).toMatchSnapshot()
  })
})
