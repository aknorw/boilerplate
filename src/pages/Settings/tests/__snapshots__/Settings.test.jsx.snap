// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings /> should render correctly 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Settings"
  />
  <h1>
    Settings
  </h1>
  <Button
    onClick={[MockFunction]}
  >
    Change foo from Settings
  </Button>
</div>
`;
